frontend/frontend/src/components/Common/RacerBarChart.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <div class="bar-chart">
+      <div class="chart-container">
+        <canvas id="racer-bar-chart"></canvas>
+      </div>
+  </div>
+</template>
+
+<script>
+import { Chart as ChartJS, BarController, BarElement, CategoryScale, LinearScale } from 'chart.js'
+ChartJS.register(BarController, BarElement, CategoryScale, LinearScale)
+
+export default {
+  props: {
+    chartConfig: Object,
+    initialConditions: Array, //Listed in the order displayed sequentially in chart
+    simRunning: Boolean,
+    simData: Array,
+    simMaxVal: Number
+  },
+  data() {
+    return {
+      racerChart: null
+    };
+  },
+  watch: {
+    initialConditions: function(value) { //Update initial bar lengths upon user param change
+      for(let i = 0; i < value.length; i++) {
```

Comment:
```suggestion
    initialConditions: function(dataPoints) { //Update initial bar lengths upon user param change
      for(let i = 0; i < dataPoints.length; i++) {
```

Use a descriptive name